### ISSUE TITLE
Understanding Demand

### ISSUE THREAD

**Issue by aesculus** [PeteRager/lennoxs30api#66]
I know that I asked this question in item #62 about what Demand was and the response was CFM. But I don't think that is correct. I have never seen a Demand that exceeded 100. Mostly for me its about 35-45, but can go up to 100. My actual CFM limits are much larger (see below).

I am wondering if it actually is the percent that is set for the zone? If you have a zoned system you set each of the zones airflow in the Test section. Don't get me started why they are there.  You can set these for (in CFM):

- Heating
- Cooling
- Blower (I guess for fan only use)

The maximum is based on the air handler you have and the installer decides how much they want each zone to get at a maximum.

Looking over the s30api_async.py file I did not see any place to obtain these fan set points. So if my hunch about percent is correct, then there seems to be no way, other than manually entering the values, to actually determine the CFM for a zone at a given time.

**Comment by PeteRager** [comment]
The configuration data for the zone CFM is probably in the "parameters" of the equipment.  Since this is metadata driven there are no predefined elements.  Once you know the PID then you can get or set that data.

It sounds like the zone demand attribute for a zones system is the % of the CFM allocated to the zone whereas in an unzoned system it is the CFM of the blower.   I have not looked at that element in a while because the equipment diagnostics provide a set of CFM related data.

**Comment by aesculus** [comment]
I have one zoned and one un-zoned. The un-zoned has not been on since I started data collection so I will report back what I find next time it fires up.

Since the airflow maxes are operation dependent and fixed, I think what I will do is to capture them in Grafana variables and then multiply against the Demand return via the API for the dashboard. And if it turns out the non zoned one is CFM its variables will all be 1.

**Comment by aesculus** [comment]
Did a quick experiment: Single zone S30 was set to 78F with ambient at 75F, heating mode. Unit came on and after a few minutes set the Demand at 50. After about 10 minutes I increased the temperature to 80F and then after a few minutes the Demand as 100 and stayed that way for the 10 minutes I ran it.

So in my case both my zoned and un-zoned S30's seem to report the percent of max airflow the unit is configured for. In the advanced settings for the none zoned S30 under Heat Pump I can see there are both low and high settings for heating, cooling and blower.

So maybe when the S30 is connected to a heat pump the Demand reports the current percent of airflow for that operation. Perhaps other type of systems report direct airflow.
